gapic/src/main/com/google/gapid/util/Crash2ExceptionHandler.java

```diff
@@ -0,0 +1,106 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.gapid.util;
+
+import static com.google.gapid.util.GapidVersion.GAPID_VERSION;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.MultipartBuilder;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.RequestBody;
+import com.squareup.okhttp.Response;
+import okio.Buffer;
+import com.google.common.base.Throwables;
+
+/**
+ * Handles uncaught exceptions and sends stacktraces to Crash2 server.
+ */
+public class Crash2ExceptionHandler implements Thread.UncaughtExceptionHandler {
+  private static final Logger LOG = Logger.getLogger(Crash2ExceptionHandler.class.getName());
+
+  // TODO(baldwinn860): Send to production url when we get approval.
+  private static final String CRASH_REPORT_URL_BASE = "https://clients2.google.com/cr/staging_report?";
+  private static final String CRASH_REPORT_PRODUCT = "GAPID";
+  private static final String CRASH_REPORT_VERSION = "Client:" + GAPID_VERSION.toString();
+  private static final String CRASH_REPORT_VERSION_ENCODED;
+  private static final String CRASH_REPORT_URL_FIELDS;
+  private static final String CRASH_REPORT_URL;
+  static { 
+    String temp;
+    try {
+      temp = URLEncoder.encode(CRASH_REPORT_VERSION, "UTF-8");
+    } catch(UnsupportedEncodingException e) {
+      temp = "UnknownVersion";
+    }
+    CRASH_REPORT_VERSION_ENCODED = temp;
+    CRASH_REPORT_URL_FIELDS = "product=" + CRASH_REPORT_PRODUCT + "&version=" + CRASH_REPORT_VERSION_ENCODED;
+    CRASH_REPORT_URL = CRASH_REPORT_URL_BASE + CRASH_REPORT_URL_FIELDS;
+  }
+
+  private final Thread.UncaughtExceptionHandler previousHandler;
+
+  public static void Register() {
+    if (!(Thread.getDefaultUncaughtExceptionHandler() instanceof Crash2ExceptionHandler)) {
+      Crash2ExceptionHandler handler = new Crash2ExceptionHandler(Thread.getDefaultUncaughtExceptionHandler());
+      Thread.setDefaultUncaughtExceptionHandler(handler);
+    }
+  }
+
+  public Crash2ExceptionHandler(Thread.UncaughtExceptionHandler oldHandler) {
+    previousHandler = oldHandler;
+  }
+
+  @Override
+  public void uncaughtException(Thread thread, Throwable thrown) {
+    Scheduler.EXECUTOR.submit(new Runnable() {
```

Comment:
Since you don't return a result and ignore the returned future, simply use `execute`.

Reply:
Also, because this can happen at any time, ignore my previous suggestion to use Scheduler.EXECUTOR and just create a thread manually.

Do we need some rate control?